src/ImageSharp/Common/Helpers/SimdUtils.HwIntrinsics.cs

```diff
@@ -644,7 +644,7 @@ public static class HwIntrinsics
             ReadOnlySpan<byte> source,
             Span<float> dest)
         {
-            fixed (byte* sourceBase = source)
+            fixed (byte* sourceBase = &MemoryMarshal.GetReference(source))
```

Comment:
This avoids the length-check.

```c#
fixed (byte* sourceBase = source)
```
is compiled to (via pattern matching)
```c#
fixed (byte* ptr = &span.GetPinnableReference())
```
and `Span<T>.GetPinnableReference` does a length-check which we don't need here, so we can omit it.

See [sharplab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgCUBXAOwwEt8YLAMIR8AB14AbGFADKMgG68wMXAG4a9Jq049+ggJI8ZEMfKhKV6mpoDMDbrmwAzGE1IMhDGgG8aDAKZ7YhQGAFlyAAo2GGwAEwB5LkkAT1kxbC4AHmAUjBgAPgZcDK4ASm9qQIY/KurA514EGDiGSNz8gCoGMQwoBgBeYtKy/3qA2vHxgDEICEjeqDKNOvGAXzGAjdXK6uJg0LDSaNjE5LTSnLzC4cyKzZqH6sbm1vbr7sXBhgAyMJh8NAUmFsFBcAALbCSFgAcRgGBirlgXBUkRKd1GO2qkym1Vm80WyyegW21VJgQeAG1/hhwRA4gZxJJIjS6QymQlerwIFxcCwAHIQIySXhcUUAczKAF0HmIoLwFNh8kxQvj3l0en1Mdipg9tmsgA=) for example.

Reply:
Do such (out-of-hot-loop) length checks have a measurable impact on perf? The new code feels bloated, I'm not in favor of such changes if there are no tangible gains.

Reply:
For this method in HwIntrinsics I don't think it's really measurable in (micro-) benchmarks. But it reduces the machine code a bit and the length-check is just not needed at all here.

I'm more on the eager side to squeeze out unnecessary instructions.
But when you want me this change to revert I'm fine with that too.

> The new code feels bloated

The C# code is bloated. That's the obvious downside.

Reply:
We typically call these methods for entire image scanlines (->100's or 1000's of elements in the span). I would assume an extra bounds check at the beginning of the method is negligible, unless there are benchmarks proving it otherwise.

> The C# code is bloated.

When you turn it into semi-C, yes. But you don't always have to; we need to make informed, individual decisions.

Reply:
OK - I'll revert.